tests/backup/backup_helper.go

```diff
@@ -1242,7 +1246,377 @@ func backupSuccessCheck(backupName string, orgID string, retryDuration time.Dura
 	return nil
 }
 
-// restoreSuccessCheck inspects restore task
+// ValidateBackup validates a backup and returns a clone of the provided scheduledCtxs (and each of their specs) *after* filtering the specs to only include the resources that are in the backup.
+// * An error can be returned *with* backedupAppContexts, so do check if the first variable returned is actually nil
+// *
+// * # Parameters
+// *
+// * requireAllScheduledCtxAreInBackup: This requires that Backup objects are a superset of objects in ScheduledCtx: If set to true, error is returned if something in scheduledCtx is not present in BackupCtx (set to false if not everything in scheduledCtx has been backed up, on purpose)
+// * requireAllBackupAreInSomeScheduledCtx: This requires that ScheduledCtx objects are a superset of objects in Backup: If set to true, error is returned if something in Backup is not present in scheduledCtx. (set to false if deploying CRs)
+// * NOTES: Setting both to true will check for strict equality of both sets. Set both to false if you just want to filter and get whatever is possible.
+// *
+// * NOTE: Ensure scheduledCtxs which correspond to the namespaces backed up in the backup are provided
+// * NOTE: This function must be called after the backup is completed (with status Success/PartialSuccess)
+func ValidateBackup(ctx context.Context, backupName string, orgID string, scheduledAppContexts []*scheduler.Context, requireAllScheduledCtxAreInBackup, requireAllBackupAreInSomeScheduledCtx bool, backupClusterConfigPath string) ([]*scheduler.Context, error) {
```

Comment:
Remove the parameters - `requireAllScheduledCtxAreInBackup, requireAllBackupAreInSomeScheduledCtx bool, backupClusterConfigPath string`

Reply:
too many parameters is not, see if you can reduce or pass some backup struct will all the fields